### PR TITLE
Space Geese NOW

### PR DESCRIPTION
**Pull request by TheFriggenSawbones**
It's happening.

This pr adds a new type of mob, The Space Goose.

Very similar to space carp, a bit more beefy and less damage. Quacks when it attacks and can be butchered for LIVERS which you can use to make a new dish, Foie Gras.

This PR also reworks the Carp Migration event, it's now An Animal Migration. It's basically the same thing except it can either spawn Space Geese or Carp.

Foie Gras is made by microwaving a Goose Liver and some Corn.

I swear to god I'll make space geese the greatest simple_mob in the world

~~~Credits~~~
Space Goose Sprite @Dizor 
Liver and Foie Gras sprites @Blithering 
Github help @HarseTheef and a mysterious individual 
Coding help @JustSumBody 

🆑 
 * rscadd: Adds Space Geese
 * rscadd: Adds Goose Livers
 * rscadd: Adds Foie Gras
 * rscadd: Reworks the Carp Migration event.

### CODE REVIEW

**Comment by BobdaBiscuit** [comment]
CALL THE SHUTTLE THE GOOSE IS LOOSE!

**Comment by ghost** [comment]
Congratulations!

**Comment by salamandooo** [comment]
I like it, very cute sprite

**Comment by Orangerush** [comment]
more geese sprites WHEN?

**Comment by ReindH** [comment]
That's a duck.

**Comment by FudgePucker** [comment]
11/10 :ok_hand:

**Comment by TheFriggenSawbones** [comment]
@ReindH please don't make up lies to defame me

**Comment by Shadowmech88** [comment]
>less damage than space carp

Don't underestimate the strength of geese.
Seriously, a goose will mess you up.

**Comment by AI-Is-Rogue** [comment]
@Shadowmech88 
I can confirm this. Geese are terrifying.

**Comment by IAGFPRMF** [comment]
So what do they look like on space tiles? Do they "walk" through space like in the gif above?

**Comment by carshalash** [comment]
Nigga thats just the duck from courage the cowardly dog with red eyes. That aint no goose.

image

**Comment by ReindH** [comment]
TOLD YOU SO!

**Comment by TheFriggenSawbones** [comment]
@IAGFPRMF http://i.imgur.com/cSY9ZIZ.gifv

**Comment by carshalash** [comment]
The ducks from courage the cowardly dog were also from space.

**Comment by Gunosaguki** [comment]
make them passive when honked by a clown

**Review by Anton-Kr — CHANGES_REQUESTED** [review]
ITS A FUCKING DUCK

**Comment by AutismSpider** [comment]
duck_brothers

Were  the space chicken?

**Comment by Riekland** [comment]
but
why

**Comment by TheFriggenSawbones** [comment]
why not

**Comment by carshalash** [comment]
Duck, duck, clearly not a goose

**Comment by NigglyWiggly** [comment]
The problem is that Ducks are nice and Geese are hell creatures and this is clearly just a duck with a long neck

**Comment by Exxion** [comment]
From the front, its head looks like a Bellsprout

**Comment by BestName** [comment]
@Shadowmech88
You're thinking of swans

**Comment by Shadowmech88** [comment]
Geese may not be as bad as swans, but they're still pretty bad.

**Comment by Exxion** [comment]
The only difference I'm aware of is that swans know no fear

File: code/modules/reagents/reagent_containers/food/snacks/meat.dm [hunk 1]
```diff
@@ -213,3 +213,13 @@ var/global/list/valid_random_food_types = existing_typesof(/obj/item/weapon/reag
 
 	reagents.add_reagent(OXYGEN, rand(1,5))
 	reagents.add_reagent(ETHANOL, rand(1,5))
+
+
+/obj/item/weapon/reagent_containers/food/snacks/meat/goosemeat
+	name = "goose liver"
+	desc = "The liver from the eldritch space goose."
+	icon_state = "goosemeat"
+	New()
```
Comment (DamianX) [inline code/modules/reagents/reagent_containers/food/snacks/meat.dm]: ugh, relative paths

**Comment by DamianX** [comment]
Space goose has evolved to be far greater than the traditional goose. In fact, it sports not one but FOUR livers that you can butcher out of it.
image

**Comment by TheFriggenSawbones** [comment]
the goose lore is deep

you see... NanoTrasen bio-engineered the geese to have MULTIPLE livers, and be space fairing, so they could mass produce foie gras and sell it to rich people for dope profits.

except one day the syndicate blew up Goose Observation Post 32 and released the geese into the wild, causing the great Murder of Space Geese to spread throughout the universe.

File: code/game/machinery/kitchen/microwave.dm [hunk 2]
```diff
@@ -244,6 +244,9 @@
 				display_name = "Turnovers"
 				items_measures[display_name] = "turnover"
 				items_measures_p[display_name] = "turnovers"
+			if (istype(O,/obj/item/weapon/reagent_containers/food/snacks/meat/goosemeat))
```
Comment (ComicIronic) [inline code/game/machinery/kitchen/microwave.dm]: Can you put this with the rest of the meat? It makes the block a bit neater.

File: code/modules/admin/topic.dm [hunk 3]
```diff
@@ -2871,12 +2871,12 @@
...
-			if("carp")
+			if("migration")
 				feedback_inc("admin_secrets_fun_used",1)
 				feedback_add_details("admin_secrets_fun_used","C")
-				var/choice = input("You sure you want to spawn carp?") in list("Badmin", "Cancel")
+				var/choice = input("You sure you want to spawn a migration?") in list("Badmin", "Cancel")
 				if(choice == "Badmin")
-					message_admins("[key_name_admin(usr)] has spawned carp.", 1)
+					message_admins("[key_name_admin(usr)] has spawned an animal migration.", 1)
 					new /datum/event/carp_migration
```
Comment (ComicIronic) [inline code/modules/admin/topic.dm]: Change the name of the datum.

File: code/modules/reagents/reagent_containers/food/snacks/meat.dm (hunk 1 above)
Comment (ComicIronic) [reply code/modules/reagents/reagent_containers/food/snacks/meat.dm]: Yeah, don't do this.

**Review by ComicIronic — CHANGES_REQUESTED** [review]

**Comment by ghost** [comment]
Dang c*ders

**Comment by ComicIronic** [comment]
Banned for slur.

**Comment by TheFriggenSawbones** [comment]
travis why

**Comment by TheFriggenSawbones** [comment]
what does it MEAN

**Review by ComicIronic — APPROVED** [review]

File: code/datums/helper_datums/command_alerts.dm [hunk 4]
```diff
@@ -141,8 +141,8 @@
 
 //////CARP
 
-/datum/command_alert/carp
-	name = "Carp Migration"
+/datum/command_alert/animal_migration
+	name = "Migration"
 	alert_title = "Lifesign Alert"
 
 /datum/command_alert/carp/announce()
```
Comment (ComicIronic) [inline code/datums/helper_datums/command_alerts.dm]: Didn't change this path!

**Review by ComicIronic — CHANGES_REQUESTED** [review]

File: code/modules/events/animal_migration.dm [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+	command_alert(/datum/command_alert/animal_migration)
+
+/datum/event/animal_migration/start()
+	for(var/obj/effect/landmark/C in landmarks_list)
+		if(C.name == "carpspawn")
+			if(prob(90)) //Give it a sliver of randomness
+				spawned_carp.Add(new spawn_type(C.loc))
+
+/datum/event/animal_migration/end()
+	for(var/mob/living/simple_animal/hostile/carp/C in spawned_carp)
```
Comment (ComicIronic) [inline code/modules/events/animal_migration.dm]: This won't remove Geese after the migration is over - you need to loop over a more generic type.

File: code/modules/events/animal_migration.dm [hunk 6]
```diff
@@ -0,0 +1,27 @@
+/datum/event/animal_migration
+	announceWhen	= 20
+	endWhen = 450
+	var/list/spawned_carp = list()
```
Comment (ComicIronic) [inline code/modules/events/animal_migration.dm]: Change the name of this variable.

**Review by ComicIronic — CHANGES_REQUESTED** [review]

**Review by ComicIronic — APPROVED** [review]

**Review by PJB3005 — APPROVED** [review]

**Comment by d3athrow** [comment]
quack

**Comment by Megamaw** [comment]
official goose lore + excellent animation

merge it

**Comment by SonixApache** [comment]
i fucking hate geese

**Comment by Kuranon** [comment]
I don't like how it's basically just "it's a goose but now it's here in space too". For space carps, I thought for the longest time it was actually just some weird spacefaring alien lifeform before reading the lore because it just looked so alien and that's kind of believeable in whatever weird scifi univerese SS13 is in, but literal space birds that seem to be in no way different from the actual Earth animal is just over the line for dumb. The sprite is really mediocre compared to carp as well, and  the way it literally walks through space is very unappealing to see

**Comment by Shadowmech88** [comment]
This may be using the existence of space bears as a precedent, but space bears are incredibly uncommon mobs, usually only seen during adminbus.
I agree that, for such a common mob, having it just be an Earth animal that doesn't need to breathe seems somewhat lazy.

I'd suggest either making it an actual goose that can't survive in space, in which case it would be comparable to goats, or making it look reasonably alien so that it merely resembles a goose, in the same vein as space carp.

**Comment by clusterfack** [comment]
Slippery slope with the prefix space

**Comment by yclatious** [comment]
Glad to see its shaping up to be a real thing but like others have said, you should really get a new sprite for it, this just looks like goons wacky unkillable animals, a actual fucked up goose in space would fit much better than this wacky cutesy thing, besides, geese are the spawn of satan, why is this one cute?

**Comment by chatsonf** [comment]
It's cute cause it's actually a duck with a long neck, not a goose

**Comment by Riekland** [comment]
What he said
Space carp have extra fins and teeth, space bears have stars in their coats
What makes this anything other than just a goose

**Comment by ReindH** [comment]
The fact that it's a duck?

**Comment by PJB3005** [comment]
You know I feel that at one point we need to start having quality control.

Maybe we should begin with this PR 🤔

**Comment by yclatious** [comment]
Cmon pjb, the code is decent and the only really big issues are the sprite and maybe the liver ammount,  dont be a no fun dutchman and just ask for those to be fixed before a merge.

**Comment by TheFriggenSawbones** [comment]
@PJB3005 what do you even mean

**Comment by ghost** [comment]
Brb porting dakis

**Comment by sonime** [comment]
Make it not common, like admin-only thing and I'll give it a thumbs up

**Comment by TheFriggenSawbones** [comment]
@sonime my problem with that is I don't want it to be admin spawn only

**Comment by yclatious** [comment]
If we are all going to be honest here, even if it was adminspawn only, it would still be spammed like a certain race of short stacked door fuckerson release, except instead of a certain *someone* spawning them its just be memebones.

**Comment by Blithering** [comment]
Where would they spawn in the event? in space like carp do? It might be better to add them to the hostile animal or infestation events instead, so they aren't waddling around in space instead of in your office breaking your arm with their wings and hissing at you.

**Comment by TheFriggenSawbones** [comment]
Yeah they spawn in carp areas.

I could totally make them an infestation event instead but I'd like a bit more feedback first.

**Comment by mtpowell** [comment]
No, god no.

There's no point to this, it's a snowflake mob with a snowflake ingredient. It's even weaker than carp and therefore NERFS carps by taking up their spawns.

**Comment by Exxion** [comment]
It's a meme mob except without an actual meme to at least give it some humor value
I'm sorry Sawbones but I really don't see the point

**Comment by Riekland** [comment]
I don't actually get it is there some epic twitter meme about geese or something

**Comment by Intigracy** [comment]
conflicts
memes
what next

**Comment by TheFriggenSawbones** [comment]
@Intigracy trying 2 figure out what i need to change to get it merged, outside of the conflicts ofc

**Comment by PJB3005** [comment]
I was serious when I said we need quality control.

Seriously. Make this something other than a dumb meme or I doubt it's getting merged.

**Comment by Shadowmech88** [comment]
At the very least make the space goose look like it's not just a regular goose that can breathe in space.

**Comment by SonixApache** [comment]
wipe this meme off the face of the earth
or fix the conflicts so we can merge it

**Comment by Exxion** [comment]
https://www.youtube.com/watch?v=WChTqYlDjtI

**Pull request closed without merging**